src/azure-cli/azure/cli/command_modules/ams/_params.py

```diff
@@ -108,12 +108,14 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('account_name', account_name_arg_type)
         c.argument('sp_name', name_arg_type,
                    help="The app name or app URI to associate the RBAC with. If not present, a default name like '{amsaccountname}-access-sp' will be generated.")
-        c.argument('new_sp_name', help="The new app name or app URI to update the RBAC with.")
-        c.argument('sp_password', password_arg_type,
-                   help="The password used to log in. Also known as 'Client Secret'. If not present, a random secret will be generated.")
         c.argument('role', help='The role of the service principal.', completer=get_role_definition_name_completion_list)
+        c.argument('password_display_name', password_arg_type,
+                   help="The friendly name of the password. The actual password will be an autogenerated strong password which will be displayed after the command runs.")
+        c.argument('years', help='Number of years for which the secret will be valid. Default: 1 year.', type=int)
         c.argument('xml', action='store_true', help='Enables xml output format.')
-        c.argument('years', help='Number of years for which the secret will be valid. Default: 1 year.', type=int, default=None)
+
+    with self.argument_context('ams account sp create') as c:
+        c.argument('new_sp_name', help="The new app name or app URI to update the RBAC with.")
```

Comment:
What is this for?

Reply:
If the user wants to change the sp name

Reply:
can they not run create command with new sp name (that will create a new sp right?
)